python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py

```diff
@@ -94,16 +110,28 @@ def __init__(self, model: torch.nn.Module, input_sample=None, use_ipex=False, dt
 
     @property
     def forward_args(self):
-        return [input_value.debugName() for input_value in self.model.graph.inputs()
-                if not input_value.debugName().startswith('self')]
+        return [
+            input_value.debugName()
+            for input_value in self.model.graph.inputs()
+            if not input_value.debugName().startswith("self")
+        ]
 
     def on_forward_start(self, inputs):
         return inputs
 
     def forward_step(self, *inputs):
         if self.channels_last is True:
-            inputs = tuple(map(lambda x: x.to(memory_format=torch.channels_last), inputs))
-        return self.model(*inputs)
+            if self.channels_last_available: # it mean the input_sample is not None
+                for idx, input in enumerate(inputs):
+                    if self.channels_last_available[idx]:
+                        input.to(memory_format=torch.channels_last)
+            else:
```

Comment:
why don't we just generate the `channels_last_available` here? we have a precious input sample right :)

Reply:
change to something like
```python
else:
    self.channels_last_available = generate_channels_last_available(inputs)
    for idx, input in enumerate(inputs):
        if self.channels_last_available[idx]:
            input.to(memory_format=torch.channels_last)
```